addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * Stops the API request or websocket reconnect timer
+     */
+    private void stopTimer() {
+        ScheduledFuture<?> future = scheduledFuture;
+        if (future != null) {
+            future.cancel(false);
+            scheduledFuture = null;
+        }
+    }
+
+    /**
+     * Parses the response message to the API key generation REST API.
+     *
+     * @param r The response
+     */
+    private void parseAPIKeyResponse(AsyncHttpClient.Result r) {
+        if (r.getResponseCode() == 403) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Allow authentification for 3rd party apps. Trying again in " + String.valueOf(POLL_FREQUENCY_SEC)
+                            + " seconds");
+            scheduledFuture = scheduler.schedule(() -> requestApiKey(), POLL_FREQUENCY_SEC, TimeUnit.SECONDS);
+        } else if (r.getResponseCode() == 200) {
+            ApiKeyMessage[] response = new Gson().fromJson(r.getBody(), ApiKeyMessage[].class);
+            if (response.length == 0) {
+                throw new IllegalStateException("Authorisation request response is empty");
+            }
+            config.apikey = response[0].success.username;
+            Configuration configuration = editConfiguration();
+            configuration.put(BindingConstants.CONFIG_APIKEY, config.apikey);
+            updateConfiguration(configuration);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Waiting for configuration");
+            requestFullState();
+        } else {
+            throw new IllegalStateException("Unknown status code for authorisation request");
+        }
+    }
+
+    /**
+     * Parses the response message to the REST API for retrieving the full bridge state with all sensors and switches
+     * and configuration.
+     *
+     * @param r The response
+     */
+    private @Nullable BridgeFullState parseBridgeFullStateResponse(AsyncHttpClient.Result r) {
+        if (r.getResponseCode() == 403) {
+            return null;
+        } else if (r.getResponseCode() == 200) {
+            return new Gson().fromJson(r.getBody(), BridgeFullState.class);
+        } else {
+            throw new IllegalStateException("Unknown status code for full state request");
+        }
+    }
+
+    /**
+     * Perform a request to the REST API for retrieving the full bridge state with all sensors and switches
+     * and configuration.
+     */
+    private void requestFullState() {
+        String url = BindingConstants.url(config.host, config.apikey, null, null);
+
+        AsyncHttpClient.getAsync(url, scheduler).thenApply(this::parseBridgeFullStateResponse).exceptionally(e -> {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.warn("Get full state failed", e);
```

Comment:
If this can also happen on a timeout, please reduce to debug. Only do a warn, if this is something where we should never end up unless there's a bug in the code.